digdag-core/src/main/java/io/digdag/core/workflow/WorkflowExecutor.java

```diff
@@ -852,20 +848,42 @@ private void enqueueTask(final TaskQueueDispatcher dispatcher, final long taskId
         }).or(false);
     }
 
+    private static String encodeUniqueQueuedTaskName(StoredTask task)
+    {
+        Instant updatedAt = task.getUpdatedAt();
+        return Long.toString(task.getId()) + ".t" + String.format(ENGLISH,
+                "%x%08x", updatedAt.getEpochSecond(), updatedAt.getNano());
```

Comment:
hmm...that's right. update of updated_at happens twice from last READY state to next READY state. so current code can cause duplicated records.